Refactor Software Upgrade Documentation (#793)
astronomer/docs: software/upgrade-astronomer-stable.md

---
title: 'Upgrade Astronomer Software'
sidebar_label: 'Upgrade Astronomer'
id: upgrade-astronomer-stable
description: Upgrade to a new LTS, stable, or patch version of Astronomer Software.
---

## Overview

For Astronomer Software customers, new product features are regularly made available in stable and long-term support (LTS) releases as described in [Release and Lifecycle Policy](release-lifecycle-policy.md). Patch versions of Astronomer Software with additional bug and security fixes are also released on a regular basis.

Follow this guide to upgrade to any stable or patch version of Astronomer Software. For information on new features and changes, refer to [Software Release Notes](release-notes.md).

A few notes before you get started:
- The upgrade process will not affect running Airflow tasks as long as `upgradeDeployments.enabled=false` is set in your upgrade script.

> - Updates will not cause any downtime to Astronomer services, including the Software UI, the Astro CLI, and the Houston API.

Reviewer: ```suggestion
- Updates won’t cause Astronomer services downtime. This includes the Software UI, the Astro CLI, and the Houston API.
```

## Step 1: Review Upgrade Considerations

The [Upgrade Considerations](upgrade-astronomer.md#upgrade-considerations) section of this document contains upgrade information for specific Astronomer versions. Review these notes before starting your own upgrade process.  

## Step 2: Check Permissions

> Version upgrades can be initiated only by a user with System Admin permissions. To confirm you're an Astronomer System Admin, check that you have access to the **System Admin** menu in the Software UI:

Reviewer: ```suggestion
You need Astronomer System Admin permissions to complete version upgrades. To confirm you're an Astronomer System Admin, check that you have access to the **System Admin** menu in the Software UI:
```

![System Admin panel](https://assets2.astronomer.io/main/docs/enterprise_quickstart/admin_panel.png)

You also need permissions to create Kubernetes resources. To confirm you have those permissions, run the following commands:

```sh
kubectl auth can-i create pods --namespace <your-astronomer-namespace>
kubectl auth can-i create sa --namespace <your-astronomer-namespace>
kubectl auth can-i create jobs --namespace <your-astronomer-namespace>
```

If all commands return `yes`, then you have the appropriate Kubernetes permissions.

## Step 3: Back Up Your Database

> Before you perform an upgrade, back up your Astronomer database by following recommendations from your cloud provider or making a backup request to your organization's database administrator.

Reviewer: ```suggestion
Before you perform an upgrade, back up your Astronomer database. To create a backup, follow the instructions provided by your cloud provider, or ask your organization's database administrator for assistance.
```

If there is a problem creating your Airflow Deployment, check the commander logs to troubleshoot. Here is an example of what you will be looking for:

```
2022-04-14T05:10:45 INFO Calling commander method #updateDeployment
2022-04-14T05:10:48 INFO Response from #updateDeployment: {"result":{"message":"values don't meet the specifications of the schema(s) in the following chart(s):\nairflow:\ ... },"deployment":{}}
2022-04-14T05:10:48 INFO Deployment some-deployment successfully updated

```

Make changes as needed and rerun the upgrade command from Step 7. Do not continue to Step 8 until you have successfully created a new Airflow Deployment.

## Upgrade Considerations

This topic contains information about upgrading to specific versions of Astronomer Software. This includes notes on breaking changes, database migrations, and other considerations that might depend on your use case.

### Upgrading to 0.29

#### Minimum Pre-Upgrade Version

If you are currently on a pre-0.28 version of Software, you must upgrade to Software 0.28 before upgrading to 0.29. A direct upgrade from a pre-0.28 version to 0.29 is not possible.

> Follow the standard installation guide to upgrade to Software 0.28, then repeat the process to upgrade to 0.29.

Author: To this doc? I think we're okay here

```

If you completed this upgrade before annotating the secret, you can still complete the sync with the following command:

```bash
kubectl create job -n <release-namespace> --from=cronjob/astronomer-config-syncer upgrade-config-synchronization
```

### Upgrading to Astronomer Software 0.28

#### Version Compatibility

Before upgrading to 0.28, ensure that the following software is updated to the appropriate version:

- **Kubernetes**: Your version must be 1.19 or greater. If you need to upgrade Kubernetes, contact your cloud provider or your Kubernetes administrator.
- **Airflow Images**: You must be using an Astronomer Certified Airflow image, and the version of your image must be 1.10.15 or greater.

    For example, all of the following images would work for this upgrade:

    - `quay.io/astronomer/ap-airflow:1.10.15-7-buster`
    - `quay.io/astronomer/ap-airflow:2.0.0-3-buster-onbuild`
    - `quay.io/astronomer/ap-airflow:2.0.2-buster-onbuild`
    - `quay.io/astronomer/ap-airflow:2.2.2-onbuild`

- **Helm**: Your version must be 3.6 ≤ 3.8.

> 

Author: If there's any more compatibility to know about maybe, but I think we cover everything for the upgrade here. Not worth people second guessing whether there's another compatibility concern to worry about.
